Source/AliveLibAO/Abe.cpp

```diff
@@ -9535,9 +9535,197 @@ void Abe::State_155_ElumJumpToFall_42E060()
+            );
+            field_110_state.door = DoorStates::eUnknown_5;
+            field_196_door_id = pDoorTlv->field_24_target_door_number;
+            return;
+        }
+        case DoorStates::eUnknown_5:
+        {
+            field_B2_lvl_number = gMap_507BA8.field_0_current_level;
+            field_B0_path_number = gMap_507BA8.field_2_current_path;
+            gMap_507BA8.field_1E_door = 0;
+            auto pPathDoor = static_cast<Path_Door*>(gMap_507BA8.TLV_First_Of_Type_In_Camera_4464A0(Door_6, 0));
+            field_F0_pTlv = pPathDoor;
+
+            while (pPathDoor->field_20_door_number != field_196_door_id)
+            {
+                pPathDoor = static_cast<Path_Door*>(Path_TLV::TLV_Next_Of_Type_446500(
+                    field_F0_pTlv,
+                    Door_6
+                ));
+                field_F0_pTlv = pPathDoor;
+            }
+
+            if (pPathDoor->field_26_start_state == eDoorStates::eOpen_0)
+            {
+                if (pPathDoor->field_1E_scale == 1)
+                {
+                    field_BC_sprite_scale = FP_FromDouble(0.5);
+                    field_10_anim.field_C_layer = 13;
+                    field_C6_scale = 0;
+                }
+                else
+                {
+                    field_BC_sprite_scale = FP_FromInteger(1);
+                    field_10_anim.field_C_layer = 32;
+                    field_C6_scale = 1;
+                }
+            }
+            else if (pPathDoor->field_26_start_state == eDoorStates::eClosed_1 || pPathDoor->field_26_start_state == eDoorStates::eOpening_2)
+            {
+                if (gMap_507BA8.field_0_current_level != LevelIds::eRuptureFarmsReturn_13)
+                {
+                    field_BC_sprite_scale = FP_FromDouble(0.5);
+                    field_C6_scale = 0;
+                }
+                else
+                {
+                    field_BC_sprite_scale = FP_FromInteger(1);
+                    field_C6_scale = 1;
+                }
+            }
+            field_10_anim.field_4_flags.Set(AnimFlags::eBit5_FlipX, pPathDoor->field_46_abe_direction);
+            field_A8_xpos = FP_FromInteger((field_F0_pTlv->field_14_bottom_right.field_0_x - field_F0_pTlv->field_10_top_left.field_0_x) / 2)
+                + FP_FromInteger(field_F0_pTlv->field_C_sound_pos.field_0_x);
+            MapFollowMe_401D30(TRUE);
+
+            PathLine* pLine = nullptr;
+            FP hitX = {};
+            FP hitY = {};
+            if (sCollisions_DArray_504C6C->RayCast_40C410(
+                field_A8_xpos,
+                FP_FromInteger(field_F0_pTlv->field_10_top_left.field_2_y),
+                field_A8_xpos,
+                FP_FromInteger(field_F0_pTlv->field_14_bottom_right.field_2_y),
+                &pLine,
+                &hitX,
+                &hitY,
+                field_BC_sprite_scale != FP_FromDouble(0.5) ? 7 : 0x70))
+            {
+                field_F4_pLine = pLine;
```

Comment:
might as well use field_F4_pLine  directly?

Reply:
Pushed, but I'm not sure if I like doing it so implicitly...